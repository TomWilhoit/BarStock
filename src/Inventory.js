import React, { Component } from "react";
import InventoryCat from "./InventoryCat.js";
import "./css/Inventory.scss";

class Inventory extends Component {
  constructor(props) {
    super(props);
    this.state = {
      allLiquorCats: [],
      allBeerCats: [],
      displayType: "beer",
      selectedCat: ""
    };
  }

  toggleCat = el => {
    this.setState({
      selectedCat: el
    });
  };

  populateType = () => {
    this.props.allInventory.forEach(product => {
      if (
        !this.state.allLiquorCats.includes(product.category) &&
        product.type === "liquor"
      ) {
        this.state.allLiquorCats.push(product.category);
      }
      if (
        !this.state.allBeerCats.includes(product.category) &&
        product.type === "beer"
      ) {
        this.state.allBeerCats.push(product.category);
      }
    });
  };

  toggleType = event => {
    if (this.state.displayType === "beer") {
      this.setState({
        displayType: "liquor"
      });
    } else {
      this.setState({
        displayType: "beer"
      });
    }
  };

  render() {
    this.populateType();
    let displayCategories;
    let beerClass = "toggle-select active";
    let liquorClass = "toggle-select inactive";

    if (this.state.displayType === "liquor") {
      beerClass = "toggle-select inactive";
      liquorClass = "toggle-select active";
      displayCategories = this.state.allLiquorCats;
    } else if (this.state.displayType === "beer") {
      beerClass = "toggle-select active";
      liquorClass = "toggle-select inactive";
      displayCategories = this.state.allBeerCats;
    }

    const list = displayCategories.map(category => {
      return (
        <InventoryCat
          category={category}
          key={category}
          products={this.props.allInventory}
          toggleCat={this.toggleCat}
          selectedCat={this.state.selectedCat}
          changeCart={this.props.changeCart}
          cartItems={this.props.cartItems}
        />
      );
    });

    return (
      <div className="Inventory">
        <div className="Inventory-toggles">
          <button className={beerClass} onClick={this.toggleType}>
            <h3>
              <i className="fas fa-beer" /> Beer
            </h3>
          </button>
          <button className={liquorClass} onClick={this.toggleType}>
            <h3>
              <i className="fas fa-glass-whiskey" /> Liquor
            </h3>
          </button>
        </div>
        <div className="Category-display">
          <ul>{list}</ul>
        </div>
      </div>
    );
  }
}

export default Inventory;
