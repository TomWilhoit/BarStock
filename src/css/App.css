/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE: 
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  /* background: #1d2023; */
}

body {
  height: 100%;
}

/* #endRegion Global */



.Header-container {
  padding-bottom: 2em;
}

.Footer-container {
  position: relative;
  width: 100%;
  text-align: center;
  /* background: #1d2023; */
}

