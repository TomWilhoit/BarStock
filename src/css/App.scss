/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE:
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  background: url(./dust_scratches.png);
}

body {
  height: 100%;
}

/* #endRegion Global */

#root {
  height: 100%;
}

.App {
  height: 100%;
}

.All-Content {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}


.Header-container {
  margin-bottom: 2em;
}


/* #region Content Container  */


.content-container {
  display: table;
  max-width: 1200px;
  width: 100%;
  margin: auto;
}

.Inventory-container {
  display: table-cell;
  width: 60%;
  padding: 0em 0em 0em 2em;
}

.Totals-container {
  display: table-cell;
  width: 40%;
  padding: 0em 2em 0em 2em;
  vertical-align: bottom;
}

/* #endRegion Content Container */


.Submit-order {
  button {
    width: 100%;
    border-radius: 4px;
    outline: none;
    background: #EEAD0E;
    color: #fff;
    text-transform: uppercase;
    font-weight: 100;
    font-size: 1em;
    padding: .2em;
    font-family: 'Abel', sans-serif;
    cursor: pointer;
    &.disabled {
      pointer-events: none;
      opacity: .2;
    }
    &.enabled {
      pointer-events: all;
      opacity: 1;
    }
  }
}


.Footer-container {
  width: 100%;
  text-align: center;
  margin-top: 2em;
}



@media only screen and (max-width: 800px) {
  .content-container {
    display: block;
  }
  .Inventory-container {
    margin: auto;
    display: block;
    width: 100%;
    max-width: 600px;
    padding: 0;
    padding-bottom: 2em;
  }
  .Totals-container {
    display: block;
    margin: auto;
    width: 100%;
    padding: 0;
    max-width: 600px;
  }
}
