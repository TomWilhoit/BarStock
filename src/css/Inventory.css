
.Inventory-container {}

.Inventory {
    display: flex;
    flex-direction: column;
    width: 100%;
    max-width: 600px;
    margin: auto;
    background: #ececec;
    box-shadow: 0px 1px 7px 0px rgba(35, 31, 32, .4);
}

.Inventory-toggles {
    display: flex;
    justify-content: center;
}

.toggle-select {
    text-align: center;
    width: 50%;
    color: #fff;
    padding: 1em;
    font-weight: bold;
    text-transform: uppercase;
    cursor: pointer;
    transition: all .5s ease;
    font-family: 'Abel', sans-serif;
    background: #3ebab4;
}

/* .toggle-select:hover {
    background: #cccccc;
} */

.toggle-select.inactive {
    background: #cccccc;
}


.toggle-select h3 {
    margin: 0;
    
}

.Category-display {

}

.Category-display ul {
    margin: auto;
    width: 80%;
    padding: 1em 0;
}

.Category-display ul li {
    list-style-type: none;
    text-align: center;
    padding: .5em;
    color: #fff;
    font-family: 'Roboto Condensed', sans-serif;
    background: #231f20;
    margin: .5em;
}