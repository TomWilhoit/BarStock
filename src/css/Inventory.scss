$font-title: 'Abel', sans-serif;
$btn-active: #2c4969;
$btn-inactive: #1d2023;

.Inventory {
    display: flex;
    flex-direction: column;
    width: 100%;
    background: #ececec;
    box-shadow: 0px 0px 6px 1px rgba(35, 31, 32, .2);

    .Inventory-toggles {
        display: flex;
        justify-content: center;
    }

    .toggle-select {
        text-align: center;
        width: 50%;
        color: #fff;
        padding: .75em;
        font-size: .9em;
        font-weight: bold;
        text-transform: uppercase;
        outline:none;
        cursor: pointer;
        transition: all .5s ease;
        font-family: $font-title;
        background: $btn-active;

        &.inactive {
          background: $btn-inactive;

          &:hover {
            background: $btn-active;
          }
        }

        h3 {
          margin: 0;
        }
    }
}
