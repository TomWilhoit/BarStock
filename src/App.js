import React, { Component } from 'react';
import MockData from './mockData.js'
import './css/App.css';
import Inventory from './Inventory.js';
import Totals from './Totals.js';
import Header from './Header.js';
import Footer from './Footer.js';
import Login from './Login.js';



class App extends Component {
  constructor() {
    super();
    this.state = {
      allProducts: [MockData.distributor[0].inventory], 
      allMenu: [MockData.menu],
      cartItems: [],
      totalCost: 0,
      totalProjected: 0,
      currentUser: ""
    }
  }
  toggleAlert() {
    alert('toggle');
  }
  render() {
    return (
      <div className="App">
        {/* <div className="Login-container">
          <Login/>
        </div> */}
        <div className="Header-container">
          <Header/>
        </div>
        <div className="Inventory-container">
          <Inventory allProducts={this.state.allProducts}/>
        </div>
        <div className="Totals-container">
          <Totals allProducts={this.state.allProducts} 
                  allMenu={this.state.allMenu} />
        </div>
        <div className="Footer-container">
          <Footer/>
        </div>
      </div>
    )
  }
}

export default App;
