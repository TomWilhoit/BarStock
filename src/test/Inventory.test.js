import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Inventory from './Inventory'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Inventory/>, div);
  ReactDOM.unmountComponentAtNode(div);
});
