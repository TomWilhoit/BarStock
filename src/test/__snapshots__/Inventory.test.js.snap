// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory should match snapshot when all data is passed correctly 1`] = `
<div
  className="Inventory"
>
  <div
    className="Inventory-toggles"
  >
    <button
      className="toggle-select active"
      id="0"
      onClick={[Function]}
    >
      <h3>
        <i
          className="fas fa-beer"
        />
         Beer
      </h3>
    </button>
    <button
      className="toggle-select inactive"
      id="1"
      onClick={[Function]}
    >
      <h3>
        <i
          className="fas fa-glass-whiskey"
        />
         Liquor
      </h3>
    </button>
  </div>
  <div
    className="Category-display"
  >
    <ul />
  </div>
</div>
`;
